packages/ui-contract-editor/src/lib/ContractEditor/plugins/withClauses.js

```diff
@@ -114,17 +123,48 @@ const withClauses = (editor, withClausesProps) => {
         }
       }
 
-      onClauseUpdated(clauseNode).then(({ node, operation, error }) => {
-        if (operation === 'replace_node' && node) {
-          Transforms.removeNodes(editor, { at: path });
-          Transforms.insertNodes(editor, node, { at: path });
-        } else {
+      if (onClauseUpdated) {
+        onClauseUpdated(clauseNode).then(({ node, operation, error }) => {
+          if (operation === 'replace_node') {
+            Transforms.removeNodes(editor, { at: path });
+            Transforms.insertNodes(editor, node, { at: path });
+          } else if (operation === 'update_formulas') {
+            // if we have edited a variable, then we ensure that all
+            // formulas that depend on the variable are updated based on the values in 'node'
+            if (variable && variable[0].type === VARIABLE
```

Comment:
why do we need to check `variable[0].type === VARIABLE` here? won't that always be the case because of the `match` on line 96?

Reply:
You are correct, that is not necessary.